### ISSUE TITLE
error: 'invalid_client'

### ISSUE THREAD

**Issue by ldevalbray** [Mangopay/mangopay2-nodejs-sdk#102]
Hello, I was wondering what this error meant ? I get it on every api call
I thought it was my mangopay api configured wrongly but it can't be as I verified my clientID and password and they are correct (I can sign in with those)

**Comment by mickaelpois** [comment]
Hello @ldevalbray,

You're saying you can sign in with your clientID and password: where can you do that, on our dashboard?
Credentials used to log in our Dashboard are not the same those you should use to process API calls.

The only credentials that matters to use our API are those you received when you asked for access to our andbox env. on our website, or those our support has provided when making live your production account.

Mickaël

**Comment by ldevalbray** [comment]
Yes, I am using the same credentials than those I use to login on the
dashboards. I am not sure where to find those credentials (for both sandbox
and production). Is there any possibilities to change those ?
Infact, I just changed the passwords for both the dashboard and prod
accounts (to sign in to the dashboard). And I was using those to try to
make the API calls. Apparently, they're not the good ones. But never did we
receive from you any specific password to make the API calls ...
Anyway, If I understand correctly, we're supposed to have 4 different
credentials ? 1 for dashboard prod, 1 for dashboard sandbox, 1 for api
calls prod, another for api calls sandbox ?!
Your procedure is quite complicated ... ;)
It would be easier to have only one account !

[image: Images intégrées 1]

2018-01-22 14:18 GMT+01:00 Mickaël POIS <notifications@github.com>:

> Hello @ldevalbray
> <https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fldevalbray&data=01%7C01%7Clouis.de_valbray%40kcl.ac.uk%7Cad7e42a9dddb42ee46f508d5619a9ce1%7C8370cf1416f34c16b83c724071654356%7C0&sdata=z8tqCJTJ4kgzf3WBbZ%2Br04%2BTjUbH6Jey9V%2F1Mg342lU%3D&reserved=0>
> ,
>
> You're saying you can sign in with your clientID and password: where can
> you do that, on our dashboard?
> Credentials used to log in our Dashboard are not the same those you should
> use to process API calls.
>
> The only credentials that matters to use our API are those you received
> when you asked for access to our andbox env. on our website, or those our
> support has provided when making live your production account.
>
> Mickaël
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMangopay%2Fmangopay2-nodejs-sdk%2Fissues%2F102%23issuecomment-359420685&data=01%7C01%7Clouis.de_valbray%40kcl.ac.uk%7Cad7e42a9dddb42ee46f508d5619a9ce1%7C8370cf1416f34c16b83c724071654356%7C0&sdata=QkiVcLw%2B6yRczvnNm2YdVKFDVxOkMUzwOT3lISWMhf0%3D&reserved=0>,
> or mute the thread
> <https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAQHxnA5W5Ld8UbhefQyGDPhqxTbK-x4Pks5tNIqdgaJpZM4Rk5Yf&data=01%7C01%7Clouis.de_valbray%40kcl.ac.uk%7Cad7e42a9dddb42ee46f508d5619a9ce1%7C8370cf1416f34c16b83c724071654356%7C0&sdata=AHQHvboL3bdY7SsGSBr%2Ft6Nie0t%2Bw3evcmUNA7BfQ4M%3D&reserved=0>
> .
>



-- 
Louis de Valbray

*l.devalbray@gmail.com <l.devalbray@gmail.com>*
*louis.de_valbray@kcl.ac.uk <louis.de_valbray@kcl.ac.uk>*
*06.13.39.65.86*

**Comment by dfa1234** [comment]
Sorry but this is a huge concern here.
I lost some hours to figure the problem because of you lack of explanation in the error message.
'invalid_client' but which client? The user? Myself as a client? My client? Please do not forget your are providing a service for marketplace, 'client' it's very abstract, who is client of whom?
A better approach should be **"please check your API key"**

**Comment by bidoubiwa** [comment]
It takes time to understand that API key is the same as Client password. Maybe just change the wording when creating de clientId
